vsphere/resource_vsphere_compute_cluster_test.go

```diff
@@ -304,6 +304,34 @@ func TestAccResourceVSphereComputeCluster_vsanDITEncryption(t *testing.T) {
 	})
 }
 
+func TestAccResourceVSphereComputeCluster_vsanStretchedCluster(t *testing.T) {
+	resource.Test(t, resource.TestCase{
+		PreCheck: func() {
+			RunSweepers()
+			testAccPreCheck(t)
+			testAccResourceVSphereComputeClusterPreCheck(t)
+		},
+		Providers:    testAccProviders,
+		CheckDestroy: testAccResourceVSphereComputeClusterCheckExists(false),
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceVSphereComputeClusterStretchedClusterEnabled(),
+				Check: resource.ComposeTestCheckFunc(
+					testAccResourceVSphereComputeClusterCheckExists(true),
+					resource.TestCheckResourceAttr("vsphere_compute_cluster.compute_cluster", "vsan_stretched_cluster_enabled", "true"),
+				),
```

Comment:
Would be good to check the conf values match what you set in config see: https://developer.hashicorp.com/terraform/plugin/sdkv2/testing/acceptance-tests/teststep#check-functions

Reply:
cool, that's really helpful, we're also looking for some examples of validating attribute set, seems we could use [TestCheckTypeSetElemNestedAttrs(name string, key string, values map[string]string)](https://pkg.go.dev/github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource#TestCheckTypeSetElemNestedAttrs)
Will try to add more checks.

Reply:
Have updated with the check functions